ci/check-spelling.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+source "$(dirname "$0")/lib/init.sh"
+source module lib/io.sh
+
+if [[ "${CHECK_SPELLING}" != "yes" ]]; then
+  echo "Skipping the spelling check as it is disabled for this build."
+  exit 0
+fi
+
+# This script must run from the top-level google-cloud-cpp directory.
+cd "${PROJECT_ROOT}"
+
+# This controls the output format from bash's `time` command, which we use
+# below to time blocks of the script. A newline is automatically included.
+readonly TIMEFORMAT="... %R seconds"
+
+exit_status=0
+printf "%-30s" "Running cspell:"
```

Comment:
I have a mild preference for moving this block into the `./ci/check-style.sh` script instead of having it in a separate script. The reason for this is that some of us often quickly check formatting and stuff on our changes by running `./ci/kokoro/build.sh clang-tidy ./ci/check-style.sh`, and that alerts us to formatting errors and other non-build related errors before sending a PR. If we moved this into there, then we'd catch the spelling errors early too.

Also, looking at the other checks that are performed in check-style.sh, this one seems to fit in well enough.

WDYT?

Reply:
I did as you suggested, which simplified things a bit too.